internal/common/common.go

```diff
@@ -73,6 +73,92 @@ type ReferenceValues struct {
 	BuilderImageSHA256Digests []string `toml:"builder_image_sha256_digests"`
 }
 
+// ProvenanceIR is an internal intermediate representation of data from provenances.
+// We want to map different provenances of different build types to ProvenanceIR, so
+// all fields except for `binarySHA256Digest` are optional.
+type ProvenanceIR struct {
+	binarySHA256Digest       string
+	buildCmd                 []string
+	builderImageSHA256Digest string
+}
+
+// NewProvenanceIR creates a new proveance with given optional fields.
+// Every provenancy needs to have binary sha256 digest, so this is not optional.
+func NewProvenanceIR(binarySHA256Digest string, options ...func(p *ProvenanceIR)) *ProvenanceIR {
+	provenance := &ProvenanceIR{binarySHA256Digest: binarySHA256Digest}
+	for _, addOption := range options {
+		addOption(provenance)
+	}
+	return provenance
+}
+
+// WithBuildCmd adds a build cmd when creating a new ProvenanceIR.
+func WithBuildCmd(buildCmd []string) func(p *ProvenanceIR) {
+	return func(p *ProvenanceIR) {
+		p.buildCmd = buildCmd
+	}
+}
+
+// WithBuilderImageSHA256Digest adds a builder image sha256 digest when creating a new ProvenanceIR.
+func WithBuilderImageSHA256Digest(builderImageSHA256Digest string) func(p *ProvenanceIR) {
+	return func(p *ProvenanceIR) {
+		p.builderImageSHA256Digest = builderImageSHA256Digest
+	}
+}
+
+// GetBinarySHA256Digest gets the binary sha256 digest. Throws an error if the binary sha256 digest is empty.
+func (p *ProvenanceIR) GetBinarySHA256Digest() (string, error) {
+	if p.binarySHA256Digest == "" {
+		return "", fmt.Errorf("provenance does not have a binary SHA256 digest")
+	}
+	return p.binarySHA256Digest, nil
+}
+
+// GetBuildCmd gets the build cmd. Throws an error if the build cmd is empty.
```

Comment:
```suggestion
// GetBuildCmd gets the build cmd. Returns an error if the build cmd is empty.
```
nit: I think `return` better matches Golang terminology.

Reply:
SG!